File: src/leetcode/2026/05/Solution976.java
import java.util.Arrays;

public class Solution976 {
    public int largestPerimeter(int[] nums) {
        Arrays.sort(nums);
        for (int i = nums.length - 1; i >= 2; i--) {
            if (nums[i] < nums[i - 1] + nums[i - 2]) {
                return nums[i] + nums[i - 1] + nums[i - 2];
            }
        }
        return 0;
    }

    public static void main(String[] args) {
        Solution976 solution = new Solution976();

        int[] nums1 = { 2, 1, 2 };
        int result1 = solution.largestPerimeter(nums1);
        System.out.println("Output for nums1: " + result1); // Output: 5

        int[] nums2 = { 1, 2, 1 };
        int result2 = solution.largestPerimeter(nums2);
        System.out.println("Output for nums2: " + result2); // Output: 0

        int[] nums3 = { 3, 2, 3, 4 };
        int result3 = solution.largestPerimeter(nums3);
        System.out.println("Output for nums3: " + result3); // Output: 10
    }
}

File: src/leetcode/2026/05/Solution976.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution976 {
public:
    int largestPerimeter(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        for (int i = nums.size() - 1; i >= 2; i--) {
            if (nums[i] < nums[i - 1] + nums[i - 2]) {
                return nums[i] + nums[i - 1] + nums[i - 2];
            }
        }
        return 0;
    }
};

int main() {
    Solution976 solution;

    vector<int> nums1 = {2, 1, 2};
    cout << solution.largestPerimeter(nums1) << endl;

    vector<int> nums2 = {1, 2, 1};
    cout << solution.largestPerimeter(nums2) << endl;

    vector<int> nums3 = {3, 2, 3, 4};
    cout << solution.largestPerimeter(nums3) << endl;

    return 0;
}